tests/Operation/ModifyCollectionFunctionalTest.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace MongoDB\Tests\Operation;
+
+use MongoDB\Operation\ModifyCollection;
+use MongoDB\Operation\CreateCollection;
+use MongoDB\Operation\CreateIndexes;
+
+class ModifyCollectionFunctionalTest extends FunctionalTestCase
+{
+    public function testCollMod()
+    {
+        $operation = new CreateCollection($this->getDatabaseName(), $this->getCollectionName());
+        $operation->execute($this->getPrimaryServer());
+
+        $indexes = [['key' => ['lastAccess' => 1], 'expireAfterSeconds' => 3]];
+        $createIndexes = new CreateIndexes($this->getDatabaseName(), $this->getCollectionName(), $indexes);
+        $createIndexes->execute($this->getPrimaryServer());
+
+        $keyPattern = (object) ['lastAccess' => 1];
+        $modifyCollection = new ModifyCollection(
+                                $this->getDatabaseName(),
+                                $this->getCollectionName(),
+                                ['index' => ['keyPattern' => $keyPattern, 'expireAfterSeconds' => 1000]],
+                                ['typeMap' => ['root' => 'array']]
```

Comment:
Just confirming: this is needed because we the operation doesn't inherit a typeMap by default, so the driver returns the root document as a `stdClass`.